LocationServices/LocationServices/Helpers/RegionSelector.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  RegionSelector.swift
+//  LocationServices
+//
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+import Foundation
+
+class AWSRegionSelector {
+    
+    static let shared = AWSRegionSelector()
+    private init() {}
+
+    func setFastestAWSRegion(apiRegions: [String], completion: @escaping (String?) -> Void) {
+        if let cachedRegion = getFastestAWSRegion(),
+           apiRegions.contains(cachedRegion) {
+            completion(cachedRegion)
+            return
+        }
+        
+        var mapping: [String: TimeInterval] = [:]
+        let dispatchGroup = DispatchGroup()
+        
+        for region in apiRegions {
+            dispatchGroup.enter()
+            let startTime = Date()
+            var request = URLRequest(url: URL(string: "https://dynamodb.\(region).amazonaws.com")!)
+            request.httpMethod = "HEAD"
+
+            let task = URLSession.shared.dataTask(with: request) { _, response, error in
+                if error == nil {
```

Comment:
What about when the network call does have errors? Should we be making a retryable call instead? Or are we ok using the first region that did return a response? What if all regions return errors?